contracts/modules/v2/StakingPool.sol

```diff
@@ -335,7 +420,7 @@ contract StakingPool is IStakingPool, ERC721 {
       uint availableShares = _stakeSharesSupply;
 
       for (uint i = _firstActiveGroupId; i < _firstAvailableGroupId; ++i) {
```

Comment:
I don't understand this loop. Why are you using ++i? If you just want to skip _firstActiveGroupId you can just write it as `i = _firstActiveGroupId + 1`. However I believe your intention here was to subtract everything until _firstAvailableGroupId.

Also firstActiveGroupId and firstAvailableGroupId are quite confusing for me. Assuming we'd rename groups to terms as it more accurately describes it's temporal nature (starts at certain date and expires), I'd suggest also renaming them as follows:
firstActiveGroupId -> currentTermId (always determined by the current block timestamp)
firstAvailableGroupId -> firstIncludedTermId (determined by the period and grace period)
GROUP_SIZE -> TERM_DURATION (something that's denominated in days or seconds can hardly be referred to as size, but rather duration)

I also recommend documenting what each property from each struct does, I struggle following the code and reviewing it correctly without having something that reinforces the less obvious semantics.

Reply:
> Why are you using ++i?

As explained on the call/chat, the `++i` behaves identical to `i++` because the 3rd block of code from the for loop is executed after the loop.

> firstActiveGroupId and firstAvailableGroupId are quite confusing

Active means the group is not expired and the term is related to `activeStake`. In the context of the `allocateStake` function where we have a cover period, we have to additionally filter out the groups that are active but will expire before the cover would expire, so that's why the term `available` was introduced. Alternative adjectives would be "usable" or "applicable", resulting in either `firstUsableGroupId` or `firstApplicableGroupId` - let me know whichever you'd prefer.

> firstActiveGroupId -> currentTermId

Using `current` in this context would bring more confusion because there's more than just one "current" group; there are multiple active groups at the same time:
```
group id     active period
n            ======
n+1          ============
n+2          ==================
```

Reply:
Right, but only the NXM from the terms that expire after the cover ends are actually booked. The rest of them are skipped and just subject to burns. So I think `firstActiveTermId` is fine.